modules/platform/nuxeo-binary-metadata/src/test/java/org/nuxeo/binary/metadata/test/TestBinaryMetadataService.java

```diff
@@ -117,6 +117,17 @@ public void itShouldExtractSingleKeywordToStringList() throws PropertyException,
 
     }
 
+    @Test
+    public void testNullFilename() throws IOException {
+        File file = FileUtils.getResourceFileFromContext("data/iptc_one_keyword.jpg");
+        Blob blob = Blobs.createBlob(file);
+        blob.setFilename(null);
+        DocumentModel doc = session.createDocumentModel("/folder", "file", "File");
+        doc.setPropertyValue("file:content", (Serializable) blob);
+        binaryMetadataService.writeMetadata(doc, "IPTC-ONE-KW");
+        // test ok if we haven't crashed here
```

Comment:
Sonar will complain if there is no assertion... worth asserting the metadata was updated maybe.

Reply:
Really? I don't have these warnings, and I'm using the Nuxeo profile linked to SonarCloud.